keps/sig-scheduling/2458-node-resource-score-strategy/README.md

```diff
@@ -0,0 +1,676 @@
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+Define a new `ScoringStrategy` type as follows:
+
+```go
+
+type StrategyType string
+
+const (
+    LeastAllocated StrategyType = "LeastAllocated"
+    MostAllocated StrategyType = "MostAllocated"
+    RequestedToCapacityRatio StrategyType = "RequestedToCapacityRatio"
+)
+
+type ScoringStrategy struct {
+    metav1.TypeMeta
+
+    // Strategy selects which strategy to run.
+    Strategy StrategyType
+    
+    // Resources to consider when scoring.
+    // The default resource set includes "cpu" and "memory" with an equal weight.
+    // Allowed weights go from 1 to 100.
+    Resources []ResourceSpec
+
+    // Arguments specific to RequestedToCapacityRatio strategy.
+    RequestedToCapacityRatio *RequestedToCapacityRatio
+}
+
+type RequestedToCapacityRatio struct {
+    // Points defining priority function shape
+    Shape []UtilizationShapePoint
+}
+
+// Note that the two types defined below already exist in the scheduler's component config API.
+// ResourceSpec represents a single resource.
+type ResourceSpec struct {
+    // Name of the resource.
+    Name string
+    // Weight of the resource.
+    Weight int64
+}
+
+// UtilizationShapePoint represents a single point of a priority function shape.
+type UtilizationShapePoint struct {
+    // Utilization (x axis). Valid values are 0 to 100. Fully utilized node maps to 100.
+    Utilization int32
+    // Score assigned to a given utilization (y axis). Valid values are 0 to 10.
+    Score int32
+}
+
+```
+
+Add `ScoringStrategy` to the existing `NodeResourcesFitArgs`:
+
+```go
+
+// NodeResourcesFitArgs holds arguments used to configure the NodeResourcesFit plugin.
+type NodeResourcesFitArgs struct {
+    metav1.TypeMeta
+
+    // IgnoredResources is the list of resources that NodeResources fit filter
+    // should ignore.
+    IgnoredResources []string
+    // IgnoredResourceGroups defines the list of resource groups that NodeResources fit 
+    // filter should ignore.
+    // e.g. if group is ["example.com"], it will ignore all resource names that begin
+    // with "example.com", such as "example.com/aaa" and "example.com/bbb".
+    // A resource group name can't contain '/'.
+    IgnoredResourceGroups []string
+    
+    // ScoringStrategy selects the node resource scoring strategy.
+    ScoringStrategy *ScoringStrategy
+}
+```
+
+
+As of writing this KEP, scheduler component config is v1beta1. The plugins we plan to
+deprecate will continue to be configurable in v1beta1, and will not be available in v1beta2. 
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+- Unit and integration tests covering the new configuration path.
```

Comment:
Hi @ahg-g, would you be able to expand this a bit more? A little more detail here would really help. You can take a look at the comment of what test plan should consider in the KEP template as well.

Reply:
there is nothing particularly challenging to test here, we are proposing a new config parameter to an existing logic, so this doesn't really include complex interactions; I added some clarification.